contrib/natss/pkg/dispatcher/dispatcher.go

```diff
@@ -211,21 +213,34 @@ func (s *SubscriptionsSupervisor) UpdateSubscriptions(channel *eventingv1alpha1.
 		chMap = make(map[subscriptionReference]*stan.Subscription)
 		s.subscriptions[cRef] = chMap
 	}
+	var errStrings []string
 	for _, sub := range subscriptions {
+		subscriberStatus := ToSubscriberStatus(&sub, corev1.ConditionFalse, "internal error")
 		// check if the subscription already exist and do nothing in this case
 		subRef := newSubscriptionReference(sub)
 		if _, ok := chMap[subRef]; ok {
 			activeSubs[subRef] = true
 			s.logger.Sugar().Infof("Subscription: %v already active for channel: %v", sub, cRef)
+			subscriberStatus.Ready = corev1.ConditionTrue
+			subscriberStatus.Message = fmt.Sprintf("Subscription: %v already active for channel: %v", sub, cRef)
 			continue
 		}
 		// subscribe
 		natssSub, err := s.subscribe(cRef, subRef)
 		if err != nil {
-			return err
+			subscriberStatus.Ready = corev1.ConditionTrue
+			subscriberStatus.Message = err.Error()
+			errStrings = append(errStrings, err.Error())
 		}
 		chMap[subRef] = natssSub
```

Comment:
this should be part of an else statement. Otherwise, even with an error, you are always setting the condition to true...

Reply:
this method is kind of hard to follow, and I think it might be prone to bugs later when a new person has to change it... 
it'd be great if you can add some comments along the way, why things are being done, and what is being done... there are a bunch of if, else, a continue, etc. etc.

Reply:
agree. Fixed the issues. WIll put comments